controllers/admin/AdminAdyenOfficialPrestashopLogFetcherController.php

```diff
@@ -0,0 +1,284 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen PrestaShop plugin
+ *
+ * @author Adyen BV <support@adyen.com>
+ * @copyright (c) 2021 Adyen B.V.
+ * @license https://opensource.org/licenses/MIT MIT license
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ */
+
+// This class is not in a namespace because of the way PrestaShop loads
+// Controllers, which breaks a PSR1 element.
+// phpcs:disable PSR1.Files.SideEffects, PSR1.Classes.ClassDeclaration
+
+use Adyen\PrestaShop\application\VersionChecker;
+use Adyen\PrestaShop\exception\GenericLoggedException;
+use Adyen\PrestaShop\exception\MissingDataException;
+use Adyen\PrestaShop\service\adapter\classes\ServiceLocator;
+use PrestaShop\PrestaShop\Adapter\CoreException;
+use PrestaShopBundle\Utils\ZipManager;
+
+require_once _PS_ROOT_DIR_ . '/modules/adyenofficial/vendor/autoload.php';
+
+class AdminAdyenOfficialPrestashopLogFetcherController extends ModuleAdminController
+{
+    /** @var \Adyen\PrestaShop\service\adapter\classes\Configuration $configuration */
+    private $configuration;
+
+    /** @var Adyen\PrestaShop\infra\Crypto */
+    private $crypto;
+
+    /** @var VersionChecker */
+    private $versionChecker;
+
+    /** @var string $logsDirectory */
+    private $logsDirectory;
+
+    /**
+     * AdminAdyenPrestashopCronController constructor.
+     *
+     * @throws CoreException|PrestaShopException
+     */
+    public function __construct()
+    {
+        $this->configuration = ServiceLocator::get('Adyen\PrestaShop\service\adapter\classes\Configuration');
+        $this->crypto = ServiceLocator::get('Adyen\PrestaShop\infra\Crypto');
+        $this->versionChecker = ServiceLocator::get('Adyen\PrestaShop\application\VersionChecker');
+        if ($this->versionChecker->isPrestaShop16()) {
+            $this->logsDirectory = _PS_ROOT_DIR_ . '/log/adyen';
```

Comment:
is it also possible to include the other logs as well (and exceptions) that are outside of the adyen logs? (in case the platform throws an exception it might be caught by prestashop itself and not us so we won't see those errors in our logs)

Reply:
https://github.com/Adyen/adyen-prestashop/pull/185/commits/e56a22fa8010728472e694788c07a3bdcc079c20